### PR TITLE
Convergence b2c update

### PR DESCRIPTION
**Pull request by omercs**
policy parameter support
moq framework to test some policy behaviors
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/211%23issuecomment-116801617%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/211%23issuecomment-117744636%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/211%23discussion_r33698736%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/211%23discussion_r33698983%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/211%23discussion_r33797782%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/211%23discussion_r33798041%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/211%23issuecomment-116801617%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Ping....%22%2C%20%22created_at%22%3A%20%222015-06-29T19%3A18%3A56Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/466805%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/omercs%22%7D%7D%2C%20%7B%22body%22%3A%20%22Ping%22%2C%20%22created_at%22%3A%20%222015-07-01T16%3A45%3A09Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/466805%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/omercs%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%20829eabb69e2626cf91040798a58d3609102a5194%20src/ADAL.PCL/TokenCacheKey.cs%2053%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/211%23discussion_r33698736%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22can%20you%20confirm%20if%20policy%20is%20not%20case%20sensitive%3F%22%2C%20%22created_at%22%3A%20%222015-07-01T16%3A51%3A23Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/5713915%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kpanwar%22%7D%7D%2C%20%7B%22body%22%3A%20%22We%20checked%20that%20it%20is%20not%20case%20sensitive.%22%2C%20%22created_at%22%3A%20%222015-07-02T16%3A45%3A55Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/466805%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/omercs%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20src/ADAL.PCL/TokenCacheKey.cs%3AL133-140%22%7D%2C%20%22Pull%20829eabb69e2626cf91040798a58d3609102a5194%20src/ADAL.PCL/TokenCacheKey.cs%2074%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/211%23discussion_r33698983%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22only%20string.compare%20should%20do.%20https%3A//msdn.microsoft.com/en-us/library/84787k22%28v%3Dvs.110%29.aspx%22%2C%20%22created_at%22%3A%20%222015-07-01T16%3A54%3A00Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/5713915%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kpanwar%22%7D%7D%2C%20%7B%22body%22%3A%20%22This%20is%20the%20usage%20in%20the%20code%20for%20other%20compare%20as%20well.%20%22%2C%20%22created_at%22%3A%20%222015-07-02T16%3A48%3A56Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/466805%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/omercs%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20src/ADAL.PCL/TokenCacheKey.cs%3AL171-193%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/211#issuecomment-116801617'>General Comment</a></b>
- <a href='https://github.com/omercs'></a> Ping....
- <a href='https://github.com/omercs'></a> Ping
- [ ] <a href='#crh-comment-Pull 829eabb69e2626cf91040798a58d3609102a5194 src/ADAL.PCL/TokenCacheKey.cs 53'></a> &nbsp;<b><a href='https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/211#discussion_r33698736'>File: src/ADAL.PCL/TokenCacheKey.cs:L133-140</a></b>
- <a href='https://github.com/kpanwar'></a> can you confirm if policy is not case sensitive?
- <a href='https://github.com/omercs'></a> We checked that it is not case sensitive.
- [ ] <a href='#crh-comment-Pull 829eabb69e2626cf91040798a58d3609102a5194 src/ADAL.PCL/TokenCacheKey.cs 74'></a> &nbsp;<b><a href='https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/211#discussion_r33698983'>File: src/ADAL.PCL/TokenCacheKey.cs:L171-193</a></b>
- <a href='https://github.com/kpanwar'></a> only string.compare should do. https://msdn.microsoft.com/en-us/library/84787k22(v=vs.110).aspx
- <a href='https://github.com/omercs'></a> This is the usage in the code for other compare as well.

<a href='https://www.codereviewhub.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/211?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/211?approve=1'></a>&nbsp;<a href='https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/211'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by brandwe** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/211'>Refresh</a>.<a href='#crh-update-none'></a>

**Comment by omercs** [comment]
Ping....

**Comment by omercs** [comment]
Ping

File: src/ADAL.PCL/TokenCacheKey.cs [hunk 1]
```diff
@@ -119,6 +133,7 @@ public override int GetHashCode()
                 + this.ClientId.ToLower() + Delimiter
                 + this.UniqueId + Delimiter
                 + ((this.DisplayableId != null) ? this.DisplayableId.ToLower() : null) + Delimiter
+                + ((this.Policy != null) ? this.Policy.ToLower() : null) + Delimiter
```
Comment (kpanwar) [inline src/ADAL.PCL/TokenCacheKey.cs]: can you confirm if policy is not case sensitive?

File: src/ADAL.PCL/TokenCacheKey.cs [hunk 2]
```diff
@@ -156,6 +171,22 @@ public bool ScopeIntersects(string[] otherScope)
...
+                return true;
+            }
+
+            if (string.IsNullOrEmpty(this.Policy) || string.IsNullOrEmpty(otherPolicy))
+            {
+                return false;
+            }
+
+
+            return (string.Compare(otherPolicy, this.Policy, StringComparison.OrdinalIgnoreCase) == 0);
```
Comment (kpanwar) [inline src/ADAL.PCL/TokenCacheKey.cs]: only string.compare should do. https://msdn.microsoft.com/en-us/library/84787k22(v=vs.110).aspx

File: src/ADAL.PCL/TokenCacheKey.cs (hunk 1 above)
Comment (omercs) [inline src/ADAL.PCL/TokenCacheKey.cs]: We checked that it is not case sensitive.

File: src/ADAL.PCL/TokenCacheKey.cs (hunk 2 above)
Comment (omercs) [inline src/ADAL.PCL/TokenCacheKey.cs]: This is the usage in the code for other compare as well.

**Pull request merged**